tooling/template-base/renderer/_default.page.server.tsx

```diff
@@ -73,8 +56,26 @@ async function onBeforeRender(pageContext: PageContext) {
     </ApolloProvider>
   );
 
-  const pageHtml = await renderToStringWithData(App);
+  const pageHtml = await getDataFromTree(tree);
   const apolloInitialState = apolloClient.extract();
 
-  return { pageContext: { pageHtml, apolloInitialState, documentProps, queryParams, authToken } };
-}
```

Comment:
The original implementation returns additional variables for `pageContext` (like `documentProps`, etc), not sure if these need to be ported over too.

Reply:
`queryParams` and `authToken` would possibly be needed as well. Believe the `catalog` component uses the `queryParams` returned from `getPageContext`, right? Think testing whether the catalog still functions as expected with the `queryParams` removed would answer this question.

Reply:
thanks for the inputs! after reading more on vite-plugin-ssr, if i am understanding this correctly, how page context works is as follows:

- page context is bound to server by default
- server initializes page context (in helium, page context is initialized with current user, auth token, apollo client, etc)
- expose server page content to client by hook `passToClient` in `*.page.server.js`
- customize server render behavior by hook `render` in `*.page.server.js` (in helium, it exposes additional page context `apolloInitialState`)

Currently, the `passToClient` hook is implemented to expose `authToken`, `urlParsed`, etc to the client. In `catalog` page, it refers to `urlParsed` from the page context to get url pathname and extract query params. It seems to be working as is. Here is one deployed helium app with the current changes: [sample catalog page with `urlParsed`](https://v2-standalone-app-w-localnpmr.lu-jiang.workers.dev/catalog?labels=%5B%22Learning+Preference+%22%5D&values=%5B%22Blog%22%5D). think it is good for now just to expose the additional `apolloInitialState` in the `render` hook?

Reply:
@ljiang-ti Yeah that tracks. I'm having issues viewing the sample provided though (cert error) -- are you able to deploy that against a live instance, e.g., `https://lujiang.thoughtindustries.com`

Reply:
discussed w/ @rjschill87 offline, we will target staging environment to verify this through the normal helium deployment flow.